components/automate-ui/src/app/pages/project/confirm-apply-stop-modal/confirm-apply-stop-modal.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, Input, OnChanges, Output, EventEmitter } from '@angular/core';
+import { ApplyRulesStatus } from 'app/entities/projects/project.reducer';
+import * as moment from 'moment';
+
+@Component({
+  selector: 'app-confirm-apply-stop-modal',
+  templateUrl: './confirm-apply-stop-modal.component.html',
+  styleUrls: [ './confirm-apply-stop-modal.component.scss' ]
+})
+export class ConfirmApplyStopModalComponent implements OnChanges {
+
+  @Input() visible = false;
+
+  @Input() applyRulesStatus: ApplyRulesStatus;
+
+  @Output() confirm: EventEmitter<void> = new EventEmitter();
+
+  @Output() cancel: EventEmitter<void> = new EventEmitter();
+
+  public progressValue: number;
+
+  public progressPrefixText: string;
+
+  public progressSuffixText: string;
+
+  ngOnChanges() {
+    this.updateProgress(this.applyRulesStatus);
+  }
+
+  public updateProgress(applyRulesStatus: ApplyRulesStatus): void {
+    const { percentageComplete, estimatedTimeComplete } = applyRulesStatus;
+    const now = moment();
+    const etc = moment(estimatedTimeComplete);
+    const dur = moment.duration(etc.diff(now));
+    const pad = num => num.toString().padStart(2, '0');
+    const durHours = pad(Math.floor(dur.asHours()));
+    const durMins = pad(dur.minutes());
+    const durSecs = pad(dur.seconds());
+    const durCountdown = etc.diff(now) < 0 ? '00:00:00' : `${durHours}:${durMins}:${durSecs}`;
+
+    this.progressValue = percentageComplete;
+    this.progressPrefixText = `${Math.floor(percentageComplete * 100)}% complete`;
```

Comment:
Hmm... Curious why you changed from round to floor here...? Seems like round would be accurate.

Reply:
@msorens currently 99.5 will display as 99 instead of 100 with this method, so let's leave it as is, for now, we can revisit in the future if needed.

Reply:
I don't feel strongly either way; I'd happily follow either choice. It just seemed more appropriate to me to display values like `99.6` as `99% complete` as opposed to `100% complete`.